speeduino/decoders.cpp

```diff
@@ -335,6 +337,23 @@ static inline bool IsCranking(const statuses &status) {
   return (status.RPM < status.crankRPM) && (status.startRevolutions == 0U);
 }
 
+bool engineIsRunning(uint32_t curTime) {
+  // Check how long ago the last tooth was seen compared to now. 
+  // If it was more than MAX_STALL_TIME then the engine is probably stopped. 
+  // toothLastToothTime can be greater than curTime if a pulse occurs between getting the latest time and doing the comparison
+  ATOMIC() {
+    return (toothLastToothTime > curTime) || ((curTime - toothLastToothTime) < MAX_STALL_TIME); 
```

Comment:
There is a risk that interrupt state is not restored correctly.
<img width="1256" height="569" alt="image" src="https://github.com/user-attachments/assets/a727480f-ebd8-4acd-8764-346123d3bf63" />